discovery_engine_core/ai/xayn-ai/src/coi/key_phrase.rs

```diff
@@ -835,88 +682,84 @@ mod tests {
+            ],
         );
         let config = Config::default();
 
-        let top_key_phrases_first = relevances.select_top_key_phrases(
+        let top_key_phrases_first = key_phrases.remove(
             &cois,
             usize::MAX,
             config.horizon(),
             config.penalty(),
+            config.gamma(),
         );
 
-        let top_key_phrases_second = relevances.select_top_key_phrases(
+        let top_key_phrases_second = key_phrases.remove(
             &cois,
             usize::MAX,
             config.horizon(),
             config.penalty(),
+            config.gamma(),
         );
 
         assert_eq!(top_key_phrases_first, top_key_phrases_second);
     }
 
-    mod clean_key_phrase {
-        use super::*;
-
-        #[test]
-        fn no_symbol_is_identity_letters() {
-            let s = "aàáâäąbßcçdeèéêëęfghiìíîïlłmnǹńoòóôöpqrsśtuùúüvwyỳýÿzź";
-            assert_eq!(clean_key_phrase(s), s);
-        }
+    #[test]
+    fn no_symbol_is_identity_letters() {
+        let s = "aàáâäąbßcçdeèéêëęfghiìíîïlłmnǹńoòóôöpqrsśtuùúüvwyỳýÿzź";
+        assert_eq!(clean_key_phrase(s), s);
+    }
 
-        #[test]
-        fn no_symbol_is_identity_numbers() {
-            let s = "0123456789";
-            assert_eq!(clean_key_phrase(s), s);
-        }
+    #[test]
+    fn no_symbol_is_identity_numbers() {
+        let s = "0123456789";
+        assert_eq!(clean_key_phrase(s), s);
+    }
 
-        #[test]
-        fn remove_symbols() {
-            assert_eq!(clean_key_phrase("!$\",?(){};:."), "");
-        }
+    #[test]
+    fn remove_symbols() {
+        assert_eq!(clean_key_phrase("!$\",?(){};:."), "");
+    }
 
-        #[test]
-        fn remove_symbols_adjust_space_between() {
-            for s in ["a-b", "a - b"] {
-                assert_eq!(clean_key_phrase(s), "a b");
-            }
+    #[test]
+    fn remove_symbols_adjust_space_between() {
+        for s in ["a-b", "a - b"] {
+            assert_eq!(clean_key_phrase(s), "a b");
         }
+    }
 
-        #[test]
-        fn remove_symbols_adjust_space_after() {
-            for s in ["a!  ", "a ! ", "a  !  "] {
-                assert_eq!(clean_key_phrase(s), "a");
-            }
+    #[test]
+    fn remove_symbols_adjust_space_after() {
+        for s in ["a!  ", "a ! ", "a  !  "] {
+            assert_eq!(clean_key_phrase(s), "a");
         }
+    }
 
-        #[test]
-        fn remove_symbols_adjust_space_before() {
-            for s in ["  !a ", " ! a ", "  !  a  "] {
-                assert_eq!(clean_key_phrase(s), "a");
-            }
+    #[test]
+    fn remove_symbols_adjust_space_before() {
+        for s in ["  !a ", " ! a ", "  !  a  "] {
+            assert_eq!(clean_key_phrase(s), "a");
         }
+    }
 
-        #[test]
-        fn adjust_spaces() {
-            assert_eq!(clean_key_phrase("  a  b  c  "), "a b c");
-        }
+    #[test]
+    fn adjust_spaces() {
+        assert_eq!(clean_key_phrase("  a  b  c  "), "a b c");
```

Comment:
:wrench: What's the reason to move this out of the their own mod? it is easier to identify what part this group of functions are testing and I don't see any reason to move them out in this PR.